master/internal/resourcemanagers/priority.go

```diff
@@ -383,3 +383,39 @@ func taskFilter(label string, zeroSlots bool) func(*sproto.AllocateRequest) bool
 		return request.Label == label && (request.SlotsNeeded == 0) == zeroSlots
 	}
 }
+
+func sortTasks(
+	taskList *taskList,
+	groups map[*actor.Ref]*group,
+	k8s bool,
+) []*sproto.AllocateRequest {
+	var reqs []*sproto.AllocateRequest
+
+	for it := taskList.iterator(); it.next(); {
+		reqs = append(reqs, it.value())
+	}
+
+	sort.Slice(reqs, func(i, j int) bool {
+		p1 := *groups[reqs[i].Group].priority
+		p2 := *groups[reqs[j].Group].priority
+		if k8s { // in k8s, higher priority == more prioritized
```

Comment:
nit I think you can flip p1 and p2 value like we do in api_job for a bit shorter syntax